lib/tasks/manage.rake

```diff
@@ -17,7 +17,33 @@ namespace :manage do
     end
   end
 
-  desc "Unsubscribe a single subscriber"
+  desc "Unsubscribe a subscriber from a single subscription"
+  task :unsubscribe_single_subscription, %i[email_address subscriber_list_slug] => :environment do |_t, args|
+    email_address = args[:email_address]
+    subscriber = Subscriber.find_by_address(email_address)
+    subscriber_list = SubscriberList.find_by(slug: subscriber_list_slug)
+    if subscriber.nil?
+      puts "Subscriber #{email_address} not found"
+    elsif subscriber_list.nil?
+      puts "Subscriber list #{subscriber_list_slug} not found"
+    elsif !(subscriber.subscriptions.map(&:subscriber_list_id).include? subscriber_list.id)
+      puts "Subscriber #{email_address} does not appear to be signed up for #{subscriber_list_slug}"
+    else
```

Comment:
We could probably slim down this else block to whats below and do most of the work in a query:

```
active_subscriptions = Subscription.active.where(subscriber_list: subscriber_list, subscriber: subscriber)

puts "Unsubscribing from #{email_address} from #{subscriber_list_slug}"
active_subscriptions.each { |s| UnsubscribeService.subscription!(s, :unsubscribed) }
```

What do you think?

Reply:
Nice 👍 I like it! I'll just do a `.last` to make it `active_subscription` (singular), as there should never be more than 1 active subscription to the same subscriber list.

Reply:
This gets us from 12 lines to 9, although we lose the end date & reason:

```ruby
    else
      active_subscriptions = Subscription.active.where(subscriber_list: subscriber_list, subscriber: subscriber)
      if active_subscriptions.empty?
        puts "Subscriber #{email_address} already unsubscribed from #{subscriber_list_slug}"
      else
        UnsubscribeService.subscription!(active_subscriptions.last, :unsubscribed)
        puts "Unsubscribing from #{email_address} from #{subscriber_list_slug}"
      end
    end
```

What do you think? Worth doing?

Reply:
Getting the 'last' active subscription sounds fine as long as it's not possible to have multiple otherwise we could risk not unsubscribing someone.

Reply:
Yup it looks an improvement to me 👍I'm not sure the reason and date are important as if you're running this task you're unlikely to care about ended subscriptions anyway.

Reply:
Done 👍 Agreed.